docs/user_documentation.md

```diff
@@ -78,20 +78,22 @@ Then start tensorboard and point it to the model directory (or any parent direct
 ### CPU/GPU training
 
 By default, training is carried out on the first GPU device of your machine.
-You can specify alternative devices with the `--device-ids` option, with
+You can specify alternative GPU devices with the `--device-ids` option, with
 which you can also activate multi-GPU training (see below). If
 `--device-ids -1`, sockeye will try to find a free GPU on your machine and block
-until one is available. The locking mechanism is based on files and therefore assumes all processes are running
+until all of them are available. The locking mechanism is based on files and therefore assumes all processes are running
```

Comment:
Unclear, specially the "all of them" as in the example you set it to -1. I would write something along the lines of "If you specify a negative number, sockeye will try to allocate this number of given GPUs, blocking until they become available."